src/sentry/options/manager.py

```diff
@@ -369,3 +450,47 @@ def get_last_update_channel(self, key: str) -> Optional[UpdateChannel]:
         # be applied evaluating all the possible drift cases.
         opt = self.lookup_key(key)
         return self.store.get_last_update_channel(opt)
+
+    def can_update(self, key: str, value, channel: UpdateChannel) -> Optional[NotWritableReason]:
+        """
+        Return the reason the provided channel cannot update the option
+        to the provided value or None if there is no reason and the update
+        is allowed.
+        """
+
+        required_flag = WRITE_REQUIRED_FLAGS.get(channel)
+        opt = self.lookup_key(key)
+        if opt.flags & (FLAG_NOSTORE | FLAG_IMMUTABLE):
+            return NotWritableReason.READONLY
+        if opt.flags & FLAG_PRIORITIZE_DISK and settings.SENTRY_OPTIONS.get(key):
+            return NotWritableReason.OPTION_ON_DISK
+
+        if required_flag and not (opt.flags & required_flag):
+            return NotWritableReason.CHANNEL_NOT_ALLOWED
```

Comment:
These bit-wise operators confuse the hell out of me.  Maybe a comment here saying "if the required flag is not in the set of configured flags then return not writable".  I know that's not what the underlying operation is but it describes the business case fairly well.

Reply:
replaced with a method